File: TestGen_ts/Manager.ts
import { Employee } from './Employee';

export class Manager extends Employee {
  private teamSize: number;

  constructor(name: string, salary: number, teamSize: number) {
    super(name, salary);
    this.teamSize = teamSize;
  }

  work(): void {
    console.log(`${this.name} is managing a team of ${this.teamSize} employees.`);
  }

  getDetails(): string {
    return `${super.getDetails()}, Manages: ${this.teamSize} people`;
  }
}

File: TestGen_java/Manager.java
public class Manager extends Employee {
  private int teamSize;

  public Manager(String name, double salary, int teamSize) {
      super(name, salary); // Call the constructor of Employee
      this.teamSize = teamSize;
  }

  @Override
  public void work() {
      System.out.println(getName() + " is managing a team of " + teamSize + " employees.");
  }

  @Override
  public String getDetails() {
      return super.getDetails() + ", Manages: " + teamSize + " people";
  }
}
